dspace-api/src/main/java/org/dspace/external/provider/impl/OrcidPublicationDataProvider.java

```diff
@@ -0,0 +1,471 @@
+import org.orcid.jaxb.model.v3.release.record.ExternalIDs;
+import org.orcid.jaxb.model.v3.release.record.SourceAware;
+import org.orcid.jaxb.model.v3.release.record.Work;
+import org.orcid.jaxb.model.v3.release.record.WorkBulk;
+import org.orcid.jaxb.model.v3.release.record.WorkContributors;
+import org.orcid.jaxb.model.v3.release.record.WorkTitle;
+import org.orcid.jaxb.model.v3.release.record.summary.WorkGroup;
+import org.orcid.jaxb.model.v3.release.record.summary.WorkSummary;
+import org.orcid.jaxb.model.v3.release.record.summary.Works;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Implementation of {@link ExternalDataProvider} that search for all the works
+ * of the profile with the given orcid id that hava a source other than
+ * DSpaceCris. The id of the external data objects returned by the methods of
+ * this class is the concatenation of the orcid id and the put code associated
+ * with the publication, separated by ::
+ *
+ * @author Luca Giamminonni (luca.giamminonni at 4science.it)
+ *
+ */
+public class OrcidPublicationDataProvider extends AbstractExternalDataProvider {
+
+    private final static Logger LOGGER = LoggerFactory.getLogger(OrcidPublicationDataProvider.class);
+
+    private final static int MAX_PUT_CODES_SIZE = 100;
+
+    @Autowired
+    private OrcidClient orcidClient;
+
+    @Autowired
+    private OrcidConfiguration orcidConfiguration;
+
+    @Autowired
+    private ItemService itemService;
+
+    @Autowired
+    private OrcidSynchronizationService orcidSynchronizationService;
+
+    @Autowired
+    private ImportService importService;
+
+    private OrcidWorkFieldMapping fieldMapping;
+
+    private String sourceIdentifier;
+
+    private String readPublicAccessToken;
+
+    @Override
+    public Optional<ExternalDataObject> getExternalDataObject(String id) {
+
+        if (isInvalidIdentifier(id)) {
+            throw new IllegalArgumentException("Invalid identifier '" + id + "', expected <orcid-id>::<put-code>");
+        }
+
+        String[] idSections = id.split("::");
+        String orcid = idSections[0];
+        String putCode = idSections[1];
+
+        return getWork(orcid, putCode)
+            .filter(work -> hasDifferentSourceClientId(work))
+            .filter(work -> work.getPutCode() != null)
+            .map(work -> convertToExternalDataObject(orcid, work));
+    }
+
+    @Override
+    public List<ExternalDataObject> searchExternalDataObjects(String orcid, int start, int limit) {
+        return findWorks(orcid, start, limit).stream()
+            .map(work -> convertToExternalDataObject(orcid, work))
+            .collect(Collectors.toList());
+    }
+
+    private boolean isInvalidIdentifier(String id) {
+        return StringUtils.isBlank(id) || id.split("::").length != 2;
+    }
+
+    private List<Work> findWorks(String orcid, int start, int limit) {
```

Comment:
Starting here, there are a large number of Private methods which have no comments or JavaDocs.  Please describe them, as it's not always clear what they are doing based on params.  E.g. is this returning "paginated" Works?